Model/articles.model.js

```diff
@@ -36,15 +36,44 @@ exports.updateArticleById = (article_id, inc_votes) => {
     })
 }
 
-exports.selectArticles = () => {
-    
-    return db.query(`SELECT articles.author, articles.title, articles.article_id,articles.topic, articles.created_at, articles.votes, CAST(COUNT(comments.article_id) AS int) as comment_count 
+exports.selectArticles = (sort_by = 'created_at', order = 'desc', topic) => {
+    const validSortBy = ['created_at', 'author', 'title', 'article_id', 'topic', 'votes', 'comment_count']
+
+    let qryStr = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, CAST(COUNT(comments.article_id) AS int) as comment_count 
     FROM articles 
     LEFT JOIN comments 
-        ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY created_at DESC
-    `).then(results => {
+        ON articles.article_id = comments.article_id`
+    const qryValues = []
+
+    if(topic){
+        qryStr += ` WHERE articles.topic = $1`
+        qryValues.push(topic)
+    }
+    
+    qryStr += ` GROUP BY articles.article_id`
+
+    if(sort_by){
+        if(validSortBy.includes(sort_by)) {
+         qryStr += ` ORDER BY ${sort_by}`
+        } else {
+            return Promise.reject({status: 404, msg: 'property doesnt exist'})
+        }
+    }   
+
+    if(!(order === 'asc' || order === 'desc')) {
+        return Promise.reject({status: 400, msg: 'bad request'})
+    }
+    
+
+    if(order === 'desc'){
+        qryStr += ` DESC`
+    }
```

Comment:
One more thing to think about here is if the user tries to send a query of order=DESC or order=ASC, it may be worth converting your order strings to lowercase to account for this